src/main/java/io/debezium/connector/vitess/connection/VStreamOutputMessageDecoder.java

```diff
@@ -47,8 +45,13 @@ public class VStreamOutputMessageDecoder implements MessageDecoder {
 
     private final VitessDatabaseSchema schema;
 
+    private final VitessDatabaseSchema schemaBackup;
+
     public VStreamOutputMessageDecoder(VitessDatabaseSchema schema) {
         this.schema = schema;
+        // Schema can be null. See: VitessConnector.validateConnection
+        if (schema != null) this.schemaBackup = schema.clone();
```

Comment:
There should be no cloning, both `schema` and `delayedSchema` should be passed from the constructor so it is apparent outside that two instances are needed.